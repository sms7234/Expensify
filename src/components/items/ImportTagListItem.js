import React from 'react';

export class ImportTagListItem extends React.Component {
  constructor(props){
    super(props);
    this.state={
      index: props.id,
      tag: props.Tag ? props.Tag : '',
      description:props.Description ? props.Description:'',
      buttonSave: true
    }
  }
  onTagChange = (e) => {
    const tag = e.target.value;
    this.setState(() => ({tag}));
    if (this.state.buttonSave){
      this.setState(()=>({buttonSave:false}))
    };
  };
  onDescriptionChange = (e) => {
    const description = e.target.value;
    this.setState(()=>({description}));
    if (this.state.buttonSave){
      this.setState(()=>({buttonSave:false}))
    };
  };
  onRemove = () => {
    this.props.onRemove(this.state.index);
  }
  onSave=() => {
    const update = {
      Tag: this.state.tag,
      Description: this.state.description,
    }
    this.props.onSave(update, this.state.index);
    this.setState(()=>({buttonSave:true}));
  }
  render() {
    return (
      <div>
        <div className={"group-listing " + (this.props.validation===false ? "list-item--invalid ":"")+ (this.props.validation===true ? "list-item--valid ":"")}>
          <input
            type="text"
            placeholder="tag name"
            className="text-input"
            value={this.state.tag}
            onChange={this.onTagChange}
            autoFocus
          />
          <textarea
            placeholder="Add description for your tag"
            className="textarea--multiple"
            value={this.state.description}
            onChange={this.onDescriptionChange}
          />
          <button className="button--group button--secondary" onClick={this.onRemove}>
            Remove
          </button>
          <button
            disabled={this.state.buttonSave}
            className="button button--group"
            onClick={this.onSave}>
            Save
          </button>
        </div>
      </div>
    )
  }
}

export default ImportTagListItem;
