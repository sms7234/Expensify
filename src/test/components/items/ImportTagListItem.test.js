import React from 'react';
import {shallow} from 'enzyme';
import moment from 'moment';
import ImportTagListItem from '../../../components/items/ImportTagListItem';

const data = {
  Tag: 'Groceries',
  Description: 'Text'
}

let onRemove, onSave, wrapper;

beforeEach(()=>{
  onRemove = jest.fn();
  onSave = jest.fn();
  wrapper = shallow( <ImportTagListItem
    key={'1'}
    {...data}
    id={0}
    validation={[null]}
    onRemove={onRemove}
    onSave={onSave}
  />)
})

test('test that list item renders correctly', () => {
  expect(wrapper).toMatchSnapshot();
})

test('test that remove function runs', () => {
  wrapper.find('button').at(0).simulate('click');
  expect(onRemove).toHaveBeenCalled();
});

test('test that save function runs', () => {
  wrapper.find('button').at(1).simulate('click');
  expect(onSave).toHaveBeenCalled();
  expect(wrapper.state('buttonSave')).toBe(true);
});

test('test that list item updates correctly',() => {
  const tag = "Healthcare";
  const description = "new text"
  wrapper.find('input').at(0).simulate('change', {
    target: {value: tag}
  });
  wrapper.find('textarea').simulate('change', {
    target: {value: description}
  });
  expect(wrapper.state('tag')).toBe(tag);
  expect(wrapper.state('description')).toBe(description);

  expect(wrapper.state('buttonSave')).toBe(false);
})
