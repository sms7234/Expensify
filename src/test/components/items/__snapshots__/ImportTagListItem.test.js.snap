// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test that list item renders correctly 1`] = `
<div>
  <div
    className="group-listing "
  >
    <input
      autoFocus={true}
      className="text-input"
      onChange={[Function]}
      placeholder="tag name"
      type="text"
      value="Groceries"
    />
    <textarea
      className="textarea--multiple"
      onChange={[Function]}
      placeholder="Add description for your tag"
      value="Text"
    />
    <button
      className="button--group button--secondary"
      onClick={[Function]}
    >
      Remove
    </button>
    <button
      className="button button--group"
      disabled={true}
      onClick={[Function]}
    >
      Save
    </button>
  </div>
</div>
`;
